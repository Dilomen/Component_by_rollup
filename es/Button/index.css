.button{background-color:#00f}