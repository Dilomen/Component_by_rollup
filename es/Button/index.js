import { _ as _inherits, a as _getPrototypeOf, b as _possibleConstructorReturn, c as _classCallCheck, d as _assertThisInitialized, e as _createClass } from '../getPrototypeOf-d7a15518.js';
import { _ as _defineProperty } from '../defineProperty-e8b8a880.js';
import React from 'react';
import { p as propTypes_6, a as propTypes_2, b as propTypes_7, c as propTypes_5, d as propTypes_1 } from '../index-23943bf4.js';
import { C as Classnames } from '../index-8042726d.js';
import { ConfigConsumer as Consumer } from '../configProvider/index.js';
import Icon from '../Icon/index.js';

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function () { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var Button = /*#__PURE__*/function (_React$Component) {
  _inherits(Button, _React$Component);

  var _super = _createSuper(Button);

  function Button(props) {
    var _this;

    _classCallCheck(this, Button);

    _this = _super.call(this, props);

    _defineProperty(_assertThisInitialized(_this), "renderButton", function (_ref) {
      var _Classnames;

      var getPrefixCls = _ref.getPrefixCls;
      var iconNode = null;
      var _this$props = _this.props,
          type = _this$props.type,
          size = _this$props.size,
          className = _this$props.className,
          children = _this$props.children,
          icon = _this$props.icon,
          ghost = _this$props.ghost,
          block = _this$props.block,
          onClick = _this$props.onClick,
          htmlType = _this$props.htmlType,
          href = _this$props.href,
          target = _this$props.target,
          style = _this$props.style;
      var prefixCls = getPrefixCls("btn");
      var classes = Classnames(["".concat(prefixCls)], (_Classnames = {}, _defineProperty(_Classnames, "".concat(prefixCls, "-").concat(type), type), _defineProperty(_Classnames, "".concat(prefixCls, "-").concat(size), size), _defineProperty(_Classnames, "".concat(prefixCls, "-").concat(type, "-ghost"), ghost), _defineProperty(_Classnames, "".concat(prefixCls, "-block"), block), _Classnames), className);

      if (icon) {
        iconNode = /*#__PURE__*/React.createElement(Icon, {
          icon: icon,
          className: "".concat(prefixCls, "-icon")
        });
      }

      if (href) {
        return /*#__PURE__*/React.createElement("a", {
          href: href,
          target: target,
          ref: _this.btnRef,
          onClick: onClick,
          className: classes
        }, iconNode, children);
      }

      return /*#__PURE__*/React.createElement("button", {
        type: htmlType,
        ref: _this.btnRef,
        onClick: onClick,
        className: classes
      }, iconNode, children);
    });

    _this.btnRef = React.createRef();
    return _this;
  }

  _createClass(Button, [{
    key: "render",
    value: function render() {
      return /*#__PURE__*/React.createElement(Consumer, null, this.renderButton);
    }
  }]);

  return Button;
}(React.Component);

Button.defaultProps = {
  ghost: false,
  block: false,
  htmlType: "button",
  children: "btn",
  size: "large",
  type: "default",
  className: "",
  icon: "",
  href: "",
  target: ""
};
Button.propTypes = {
  ghost: propTypes_6,
  block: propTypes_6,
  htmlType: propTypes_2(["submit", "button", "reset"]),
  children: propTypes_7,
  size: propTypes_2(["small", "middle", "large"]),
  type: propTypes_2(["default", "primary", "link", "danger"]),
  href: propTypes_5,
  target: propTypes_5,
  onClick: propTypes_1,
  icon: propTypes_7,
  className: propTypes_5
};

export default Button;
