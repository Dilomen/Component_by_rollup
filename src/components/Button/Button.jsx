import React from "react";
import "./index.scss";
import Classnames from "classnames";
import { ConfigConsumer } from "../configProvider/index.jsx";
import Icon from "../Icon/index";
import { string, bool, oneOfType, func, node, oneOf } from "prop-types";
class Button extends React.Component {
  constructor(props) {
    super(props);
    this.btnRef = React.createRef();
  }
  renderButton = ({ getPrefixCls }) => {
    let iconNode = null;
    const {
      prefixCls: customizePrefixCls,
      type,
      size,
      className,
      children,
      icon,
      ghost,
      block,
      onClick,
      htmlType,
      href,
      target,
    } = this.props;
    const prefixCls = getPrefixCls("btn", customizePrefixCls);
    const classes = Classnames(
      [`${prefixCls}`],
      {
        [`${prefixCls}-${type}`]: type,
        [`${prefixCls}-${size}`]: size,
        [`${prefixCls}-${type}-ghost`]: ghost,
        [`${prefixCls}-block`]: block,
      },
      className
    );

    if (icon) {
      iconNode = <Icon icon={icon} className={`${prefixCls}-icon`}></Icon>;
    }

    if (href) {
      return (
        <a
          href={href}
          target={target}
          ref={this.btnRef}
          onClick={onClick}
          className={classes}
        >
          {iconNode}
          {children}
        </a>
      );
    }

    return (
      <button
        type={htmlType}
        ref={this.btnRef}
        onClick={onClick}
        className={classes}
      >
        {iconNode}
        {children}
      </button>
    );
  };

  render() {
    return <ConfigConsumer>{this.renderButton}</ConfigConsumer>;
  }
}

Button.defaultProps = {
  ghost: false,
  block: false,
  htmlType: "button",
  children: "btn",
  size: "large",
  type: "default",
};
Button.propTypes = {
  ghost: bool,
  block: bool,
  htmlType: oneOf(["submit", "button", "reset"]),
  children: node,
  size: oneOf(["small", "middle", "large"]),
  type: oneOf(["default", "primary", "link", "danger"]),
  href: string,
  target: string,
  onClick: func,
  icon: oneOfType([string, node]),
};
export default Button;
