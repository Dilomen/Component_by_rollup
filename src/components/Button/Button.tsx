import React from 'react'
import './index.scss';
class Button extends React.Component {
    render() {
        return <button className="button">newqasdaswhh</button>
    }
}

export default Button