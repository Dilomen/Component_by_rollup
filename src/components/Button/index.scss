@import "../../common/style/variable.scss";
$serein-btn: #{$serein-prefix}-btn;

$color: (
  danger: $theme-danger-color,
  primary: $theme-primary-color,
  default: $theme-default-color,
);

.#{$serein-btn} {
  outline: 0;
  border-radius: 2px;
  box-sizing: border-box;
}

@each $size in small, middle, large {
  .#{$serein-btn}-#{$size} {
    @if $size == small {
      min-width: 40px;
      min-height: 24px;
      padding: 3px 8px;
      font-size: 12px;
    } @else if $size == middle {
      min-width: 45px;
      min-height: 28px;
      font-size: 14px;
      padding: 5px 10px;
    } @else {
      min-width: 50px;
      min-height: 32px;
      font-size: 16px;
      padding: 7px 12px;
    }
  }
}

@each $type in danger, primary, link, default {
  .#{$serein-btn}-#{$type} {
    cursor: pointer;
    background-color: map-get($color, #{$type});
    color: #fff;
    border: 1px solid map-get($color, #{$type});
    @if $type == default {
      border: 1px solid #d9d9d9;
      color: $text-color;
      &:hover {
        color: $theme-primary-color;
        border: 1px solid $theme-primary-color;
      }
    } @else if $type == link {
      border: 0;
      background: none;
      color: $theme-primary-color;
    }
    &:hover {
      opacity: 0.7;
    }
    &-ghost {
      background-color: transparent;
      border: 1px solid map-get($color, #{$type});
      color: map-get($color, #{$type});
      @if $type == default {
        border: 1px dashed #d9d9d9;
        color: $text-color;
        &:hover {
          color: $theme-primary-color;
          border: 1px dashed $theme-primary-color;
        }
      }
    }
  }
}

.#{$serein-btn}-block {
  display: block; 
  width: 100%;
}

.#{$serein-btn}-icon {
  margin-right: 5px;
}