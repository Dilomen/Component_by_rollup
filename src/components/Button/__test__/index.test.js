import React from "react";
import Button from "../index";
import { renderSnapshot, mountComponent } from "@tests";

mountComponent(<Button></Button>);

renderSnapshot(<Button>Basic</Button>);
renderSnapshot(<Button icon={<p>2</p>}>Icon Button</Button>);
renderSnapshot(<Button className="new-class">Icon Button</Button>)