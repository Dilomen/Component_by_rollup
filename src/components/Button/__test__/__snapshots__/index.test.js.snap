// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component could be rendered snapshots without conflict render Snapshots 1`] = `
<button
  className="serein-btn serein-btn-default serein-btn-large"
  type="button"
>
  Basic
</button>
`;

exports[`component could be rendered snapshots without conflict render Snapshots 2`] = `
<button
  className="serein-btn serein-btn-default serein-btn-large"
  type="button"
>
  <i
    className="serein-btn-icon"
  >
    <p>
      2
    </p>
  </i>
  Icon Button
</button>
`;

exports[`component could be rendered snapshots without conflict render Snapshots 3`] = `
<button
  className="serein-btn serein-btn-default serein-btn-large new-class"
  type="button"
>
  Icon Button
</button>
`;

exports[`render Snapshots asdas 1`] = `
<button
  className="serein-btn serein-btn-default serein-btn-large"
  type="button"
>
  Basic
</button>
`;

exports[`render Snapshots asdas 2`] = `
<button
  className="serein-btn serein-btn-default serein-btn-large"
  type="button"
>
  <i
    className="serein-btn-icon"
  >
    <p>
      2
    </p>
  </i>
  Icon Button
</button>
`;

exports[`render Snapshots asdas 3`] = `
<button
  className="serein-btn serein-btn-default serein-btn-large new-class"
  type="button"
>
  Icon Button
</button>
`;
